### PR TITLE
using class variables instead of constants

### PR DESCRIPTION
**Pull request by rdeutz**
# Executive summary
These change allows to make myRobo packages easier. 

# Backwards compatibility
Full b/c, no problems are expected, anyway I changed the version to 0.6.0 because this is a new feature

# Translation impact
none

# Why
In further versions of Robo I had to copy code from the runner.php to make it act and work with a custom RoboClass and custom RoboFile-Name. Now I can create my package just with setting the Robo package as dependency and create my own runner.php with extending \Robo\Runner. The 2nd step is to create a Task.php and extend \Robo\Tasks and add my tasks. So I have all what Robo has plus my tasks easily as a package available. 

I hope that does makes sense for others too. Anyone can check how I did it here (is WIP ofc): https://github.com/rdeutz/jrobo/tree/development

### CODE REVIEW

File: src/Runner.php [hunk 1]
```diff
@@ -14,22 +14,50 @@ class Runner
 {
     use IO;
 
-    const VERSION = '0.5.4';
+    const VERSION = '0.6.0';
```
Comment (boedah) [inline src/Runner.php]: The version should not be changed in a PR.

File: src/Runner.php [hunk 2]
```diff
@@ -14,22 +14,50 @@ class Runner
...
     const ROBOCLASS = 'RoboFile';
     const ROBOFILE = 'RoboFile.php';
-
+    
+    /**
+     * @var CurrentDir
```
Comment (boedah) [inline src/Runner.php]: There certainly is no class named `CurrentDir`, right?
Same for the other doc-block changes...

File: src/Runner.php (hunk 1 above)
Comment (rdeutz) [inline src/Runner.php]: find by me

**Comment by boedah** [comment]
Nice.
How does this relate to #95 and #111 ?

File: src/Runner.php (hunk 2 above)
Comment (rdeutz) [inline src/Runner.php]: I just added doc blocks didn't check it this variables are used. You are right this one and $printer aren't used at all. I will remove them and update the PR

**Comment by rdeutz** [comment]
I have seen #95 I think it is to complicated and had this idea, for me it is important to package all. We are planning to use it for the Joomla project and doing it this way allows us to support developers with some Joomla specific tasks

File: src/Runner.php (hunk 2 above)
Comment (boedah) [inline src/Runner.php]: I just actually meant the `@var` declaration and not the usage, but you seem right. Those seem to be left-overs...

**Comment by boedah** [comment]
I would say your method (where you need to create a new `Runner`) seems to be more complicated than the 3 lines of #95.
Nevertheless I think this PR is a good addition since it offers more ways to use Robo.

**Comment by DavertMik** [comment]
Sure, this makes sense, merging it then

**Pull request merged**